airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -292,20 +338,76 @@ def request_params(
         return params
 
 
+class MetafieldOrders(MetafieldShopifySubstream):
+    parent_stream_class: object = Orders
+
+
 class DraftOrders(IncrementalShopifyStream):
     data_field = "draft_orders"
 
     def path(self, **kwargs) -> str:
         return f"{self.data_field}.json"
 
 
+class MetafieldDraftOrders(MetafieldShopifySubstream):
+    parent_stream_class: object = DraftOrders
+
+
 class Products(IncrementalShopifyStream):
     data_field = "products"
 
     def path(self, **kwargs) -> str:
         return f"{self.data_field}.json"
 
 
+class MetafieldProducts(MetafieldShopifySubstream):
+    parent_stream_class: object = Products
+
+
+class ProductImages(ShopifySubstream):
```

Comment:
This class is very similar to the `OrderRefunds`, since the `product` records have `images` included within the initial response. If so, we could just make it work as `OrderRefunds` using `nested_substream = "images"`, and save the sync time for a little.